src/prefect/utilities/logging.py

```diff
@@ -33,16 +33,44 @@
 )
 
 
+class PrefectFormatter(logging.Formatter):
+    """
+    Custom formatter implementing timezone localization.
+    Creates a logger with a `StreamHandler` that has level and formatting
+    set from `prefect.config`.
+
+    Args:
+        - display_tz (str): Typically set from `prefect.config`, used by `formatTime`
+        method for display in log records.
+    """
+
+    def __init__(self, *args, display_tz: str = "UTC", **kwargs) -> None:
+        self.display_tz = display_tz
+        super().__init__(*args, **kwargs)
+
+    def converter(self, timestamp: float) -> pendulum.DateTime:
+        return pendulum.from_timestamp(timestamp, tz=self.display_tz)
+
+    def formatTime(self, record, datefmt=None) -> str:
+        dt = self.converter(record.created)
+        if datefmt:
+            s = dt.strftime(datefmt)
+        else:
+            s = dt.isoformat()
+        return s
```

Comment:
I'd generally try to avoid single letter variables and take advantage of Python's one-lining for the conditional
```suggestion
        timestr = dt.strftime(datefmt) if datefmt else dt.isoformat()
        return timestr
```
I'm not sure `timestr` is the best name choice so feel free to choose something better yet :)